### ISSUE TITLE
calcrewards.py does its own queries for approvedTokens, and shouldn't be

### ISSUE THREAD

**Issue by trentmc** [oceanprotocol/df-py#172]
query.py is supposed do all the querying in a nice clean way. It was, until recently.

Now some queries have snuck into calcrewards.py. The queries for approvedTokens. 

Problems it causes:
- Now, we need to connect across >1 chains just to calculate rewards.
- We want to see all the data grabbed from the chain sooner in the flow, so that we can inspect it, and fix queries as needed. Only once we're satisfied with the data, do we proceed to calculate rewards, and distribute them.
- In other words, it broke the clean [data flow](https://docs.google.com/presentation/d/11Nu_emSgw3iYZKDcmnvZC_aOM00MDyYTtTWQP4PYox4/edit#slide=id.p) that we developed. Now there's a weird wire from each chain to calcrewards.py
- Finally, we can't orthogonalize unit tests. Unlike before, test_calcrewards.py now depends on connecting to chains. It shouldn't need to

**Comment by trentmc** [comment]
Fixed in commit https://github.com/oceanprotocol/df-py/commit/2375b03befcb54714e1423b9e2beb93473c6efe3 as part of PR https://github.com/oceanprotocol/df-py/pull/166
